shell/platform/tizen/flutter_tizen_view.cc

```diff
@@ -0,0 +1,303 @@
+  // Compute renderer transformation based on the angle of rotation.
+  double rad = (360 - rotation_degree_) * M_PI / 180;
+  TizenWindow::Geometry geometry = window_->GetWindowGeometry();
+  int32_t width = geometry.width;
+  int32_t height = geometry.height;
+
+  double trans_x = 0.0, trans_y = 0.0;
+  if (rotation_degree_ == 90) {
+    trans_y = height;
+  } else if (rotation_degree_ == 180) {
+    trans_x = width;
+    trans_y = height;
+  } else if (rotation_degree_ == 270) {
+    trans_x = width;
+  }
+
+  flutter_transformation_ = {
+      cos(rad), -sin(rad), trans_x,  // x
+      sin(rad), cos(rad),  trans_y,  // y
+      0.0,      0.0,       1.0       // perspective
+  };
+
+  if (rotation_degree_ == 90 || rotation_degree_ == 270) {
+    std::swap(width, height);
+  }
+
+  window_->ResizeRenderTargetWithRotation(
+      {geometry.left, geometry.top, width, height}, rotation_degree_);
+
+  // Window position does not change on rotation regardless of its orientation.
+  SendWindowMetrics(geometry.left, geometry.top, width, height, 0.0);
+}
+
+void FlutterTizenView::OnPointerMove(double x,
+                                     double y,
+                                     size_t timestamp,
+                                     FlutterPointerDeviceKind device_kind,
+                                     int32_t device_id) {
+  if (pointer_state_) {
+    SendFlutterPointerEvent(kMove, x, y, 0, 0, timestamp, device_kind,
+                            device_id);
+  }
+}
+
+void FlutterTizenView::OnPointerDown(double x,
+                                     double y,
+                                     size_t timestamp,
+                                     FlutterPointerDeviceKind device_kind,
+                                     int32_t device_id) {
+  pointer_state_ = true;
+  SendFlutterPointerEvent(kDown, x, y, 0, 0, timestamp, device_kind, device_id);
+}
+
+void FlutterTizenView::OnPointerUp(double x,
+                                   double y,
+                                   size_t timestamp,
+                                   FlutterPointerDeviceKind device_kind,
+                                   int32_t device_id) {
+  pointer_state_ = false;
+  SendFlutterPointerEvent(kUp, x, y, 0, 0, timestamp, device_kind, device_id);
+}
+
+void FlutterTizenView::OnScroll(double x,
+                                double y,
+                                double delta_x,
+                                double delta_y,
+                                int scroll_offset_multiplier,
+                                size_t timestamp,
+                                FlutterPointerDeviceKind device_kind,
+                                int32_t device_id) {
+  SendFlutterPointerEvent(
+      pointer_state_ ? kMove : kHover, x, y, delta_x * scroll_offset_multiplier,
+      delta_y * scroll_offset_multiplier, timestamp, device_kind, device_id);
+}
+
+void FlutterTizenView::OnKey(Ecore_Event_Key* event, bool is_down) {
```

Comment:
This class (`FlutterTizenView`) is an abstraction layer between the platform window and Flutter, so it's expected to be platform type agnostic. Do you plan to refactor this part again in the future? Should we wrap this type (`Ecore_Event_Key`) into another type that contains all information required by TextInputChannel, PlatformViewChannel, and KeyEventChannel, or just extract the values and pass them as arguments (just like `KeyboardHook()` of the Windows embedder)?

Note) Refactoring the input system is out of scope of this PR. I just want to hear your thoughts.

Reply:
You are right. actually, I wanted to refactor handling key events similar to the point event in this PR.
However, I thought that this was related to input method context and it lead too many additional changes. so I didn't do it.
I think this is the part that must be refactored. I plan to gradually refactor Embeder including this and develop new feature related View.

Reply:
Okay, improving the input system is also related to the issue https://github.com/flutter-tizen/engine/issues/221. We'll revisit this topic soon.